internal/ir/xds.go

```diff
@@ -952,3 +961,58 @@ type TCPKeepalive struct {
 	// Defaults to `75s`.
 	Interval *uint32 `json:"interval,omitempty" yaml:"interval,omitempty"`
 }
+
+// LoadBalancer defines the load balancer settings.
+// +k8s:deepcopy-gen=true
+type LoadBalancer struct {
+	// RoundRobin load balacning policy
```

Comment:
not a blocker right now since this is just IR, but do we want to add a `type` field here?

Reply:
ah the `type` will def be in the user facing API, in this layer using the struct ptr instantiation to figure out the type

Reply:
That totally works, just wasn't sure if we wanted to make it explicitly clear in the ir that only one of these can be provided for a route, etc. for newer contributors that might be unaware. I kinda like using types in the ir sometimes.

Reply:
I added a `Validate()` method to ensure only one can be set and added test cases in this PR :)